articles/getting-started-using-admob-with-react-native-expo/index.md

```diff
@@ -0,0 +1,484 @@
+
+The overall coding implementation of this function is provided in the code snippet below:
+```jsx
+const initRewardAds = async () => {
+    // Display a rewarded ad
+    await AdMobRewarded.setAdUnitID("ca-app-pub-2547344479047582/6388494355"); // Test ID, Replace with your-admob-unit-id
+    await AdMobRewarded.requestAdAsync();
+    
+    AdMobRewarded.addEventListener("rewardedVideoDidRewardUser", () => {
+      // trigger function for hide ads
+    });
+    AdMobRewarded.addEventListener("rewardedVideoDidClose", () => {
+			// if we close ads modal will close too
+      setModalVisible(false);
+    });
+    await AdMobRewarded.showAdAsync();
+  };
+```
+
+### STEP 5: Control Ads status across app with Redux
+To control the visibility of ads across every screen in the app, we are going to use the Redux mechanism. It is similar to the previous tutorial where we control the dark theme mode changes using Redux.
+
+First, we need to define some constants in the **constant.js** file for new ad status as directed in the code snippet below:
+
+```jsx
+export const DARK_THEME = "DARK_THEME";
+export const LIGHT_THEME = "LIGHT_THEME";
+
+export const AD_ON = "AD_ON";
+export const AD_OFF = "AD_OFF";
+```
+
+Next, we need to create a new reducer file named **admob.reducer.js** and add the code to toggle the ad status as shown in the code snippet below:
+
+```jsx
+import { AD_OFF, AD_ON } from "../constant";
+const initialState = {
+  ad_status: true,
+};
+export default (ad_status = initialState, { type }) => {
+  switch (type) {
+    case AD_ON:
+      return { ad_status: true };
+    case AD_OFF:
+      return { ad_status: false };
+    default:
+      return ad_status;
+  }
+};
+```
+
+Now, we need to register the new reducer to the root reducer inside the `combineReducers` module as shown in the code snippet below:
+
+```jsx
+import { combineReducers } from "redux";
+
+import themeReducer from "./theme.reducer";
+import admobReducer from "./admob.reducer";
+export default combineReducers({
+  themeReducer,
+  admobReducer,
+});
+```
+
+Now, to trigger the reducer states, we need to create an action file named **admob.action.js**. And, add the action function to trigger the toggling of the ad status. The required action implementation is provided in the code snippet below:
+
+```jsx
+import { AD_OFF, AD_ON } from "../constant";
+export const ToggleAdOn = () => ({
+  type: AD_ON,
+});
+export const ToggleAdOff = () => ({
+  type: AD_OFF,
+});
+
+export const ToggleAds = (ads_state) => {
+  return async (dispatch) => {
+    if (ads_state === true) {
+      dispatch(ToggleAdOn());
+    } else {
+      dispatch(ToggleAdOff());
+    }
+  };
+};
+```
+
+In order to use the reducer, we need to go to the Home.js file and import the `useSelector` hook to access the reducer state easily:
+
+```jsx
+import { useSelector } from "react-redux";
+const Home = () => {
+  const admobReducer = useSelector(({ admobReducer }) => admobReducer);
```

Comment:
Even though this is 100% correct, I think the statement would be much clear if you don't restructure the argument.

```JavaScript
const admobReducer = useSelector((state) => state.admobReducer);
```

Reply:
Rewriting like this makes it clear that the argument is the state.

Reply:
Even better, you can name the variable as something like `reduxStoreState`

Reply:
updated as your mention.

Reply:
You updated it in Line 452, but forgot to update it here.

```
const admobReducer = useSelector((reduxStoreState) => state.admobReducer);
```

Reply:
@Worawat101 fix this too.

Reply:
got it I already update all of this.